### ISSUE TITLE
MA0003 warns for indexer (dictionary) access

### ISSUE THREAD

**Issue by Piedone** [meziantou/Meziantou.Analyzer#282]
**Context**
Rule identifier: _MA0003_
Target Framework: _.NET 4.6.1_
Language version: _7.3_

**Describe the bug**
MA0003 shows a violation for an indexer access of a dictionary too.

**To Reproduce**
The violation is shown on the second line:

````c#
var test = new Dictionary<bool, bool>();
var test2 = test[true];
````
(Whether it makes sense to have a `bool`-keyed dictionary is besides the point :).)

Same issue with `ILookup<TKey, TValue>`.

**Additional context**
N/A

**Comment by Piedone** [comment]
I also get the following exception:

```
118>CSC : warning AD0001: Analyzer 'Meziantou.Analyzer.Rules.NamedParameterAnalyzer' threw an exception of type 'System.IndexOutOfRangeException' with message 'Index was outside the bounds of the array.'.
```

While there are no more details this suspiciously looks related.

**Comment by meziantou** [comment]
Roslyn should display the full call stack. It should be available in the log (maybe in verbose mode or using [`/bl`](https://www.meziantou.net/exploring-the-msbuild-log-using-the-binary-log-viewer.htm)), or in the Error Window in Visual Studio (you may need to expand the line using the arrow on the left).

**Comment by Piedone** [comment]
The new release fixed both the false alarm and the exception, thank you. (And again, thank you for being super quick!)
